hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/bucket/BucketProtoUtils.java

```diff
@@ -41,12 +42,13 @@ private BucketProtoUtils() {
 
   static BucketCacheProtos.BucketCacheEntry toPB(BucketCache cache) {
     return BucketCacheProtos.BucketCacheEntry.newBuilder()
-        .setCacheCapacity(cache.getMaxSize())
-        .setIoClass(cache.ioEngine.getClass().getName())
-        .setMapClass(cache.backingMap.getClass().getName())
-        .putAllDeserializers(CacheableDeserializerIdManager.save())
-        .setBackingMap(BucketProtoUtils.toPB(cache.backingMap))
-        .build();
+      .setCacheCapacity(cache.getMaxSize())
+      .setIoClass(cache.ioEngine.getClass().getName())
+      .setMapClass(cache.backingMap.getClass().getName())
+      .putAllDeserializers(CacheableDeserializerIdManager.save())
+      .setBackingMap(BucketProtoUtils.toPB(cache.backingMap))
+      .setChecksum(ByteString.copyFrom(((PersistentIOEngine) cache.ioEngine).
```

Comment:
Not sure here, since i don't read the code. What if the cache is not PersistentIOEngine? Will here throw cast exception?

Reply:
Calling this method has shown that ioEngine is persistent, and all the persistent ioEngine extend PersistentIOEngine. So it won't throw cast exception.